pkg/scheduler/framework/frameworkutils.go

```diff
@@ -135,30 +139,32 @@ func crossReferencePickedCustersAndObsoleteBindings(
 
 		// The binding's target cluster is picked again in the current run; yet the binding
 		// is originally created/updated in accordance with an out-of-date scheduling policy.
+		// Add the binding to the toPatch list. We will simply keep the binding's state as
+		// it could be "scheduled" or "bound".
+		toPatch = append(toPatch, patchBinding(binding, binding.Spec.State, scored, policy))
+	}
 
-		// Update the binding so that it is associated with the latest score.
-		updated := binding.DeepCopy()
-		affinityScore := int32(scored.Score.AffinityScore)
-		topologySpreadScore := int32(scored.Score.TopologySpreadScore)
-		updated.Spec.ClusterDecision = placementv1beta1.ClusterDecision{
-			ClusterName: scored.Cluster.Name,
-			Selected:    true,
-			ClusterScore: &placementv1beta1.ClusterScore{
-				AffinityScore:       &affinityScore,
-				TopologySpreadScore: &topologySpreadScore,
-			},
-			Reason: pickedByPolicyReason,
+	for _, binding := range unscheduled {
+		scored, ok := pickedMap[binding.Spec.TargetCluster]
+		checked[binding.Spec.TargetCluster] = true
+		if !ok {
+			// this cluster is not picked up again, so we can skip it
+			continue
 		}
-
-		// Update the binding so that it is associated with the lastest scheduling policy.
-		updated.Spec.SchedulingPolicySnapshotName = policy.Name
-
-		// Add the binding to the toUpdate list.
-		toPatch = append(toPatch, &bindingWithPatch{
-			updated: updated,
-			// Prepare the patch.
-			patch: client.MergeFrom(binding),
-		})
+		// The binding's target cluster is picked again in the current run; yet the binding
+		// is originally de-selected by the previous scheduling round.
+		// Add the binding to the toPatch list so that we won't create more and more bindings.
+		// We need to recover the previous state before the binding is marked as unscheduled.
+		var desiredState placementv1beta1.BindingState
+		// we recorded the previous state in the binding's annotation
+		if previousState, exist := binding.GetAnnotations()[placementv1beta1.PreviousBindingStateAnnotation]; exist {
+			desiredState = placementv1beta1.BindingState(previousState)
+			// remove the annotation just to avoid confusion.
+			binding.SetAnnotations(nil)
```

Comment:
Hi Ryan! Just to be on the safe side, maybe it's better to just unset the one annotation we use for keeping the old state?